### PR TITLE
Jsfiddle LiquidTag

### PR DESCRIPTION
**Pull request by Link2Twenty**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description
Added a JSFiddle Liquid tag

## Related Tickets & Documents
#1323 

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [ ] no documentation needed
- [x] Editor guide text

### CODE REVIEW

**Comment by Link2Twenty** [comment]
@Zhao-Andy @maestromac I don't really know what `total-coverage` is, I'm guessing it's something to do with testing.
Do you know of any resources I could look up to help me make it pass those tests?

Thanks

**Comment by Link2Twenty** [comment]
@Zhao-Andy Good job you posted that, it still references codepen. I've changed it now.

**Comment by Zhao-Andy** [comment]
@Link2Twenty Ah, great catch. Funny I didn't see it either.

File: app/views/pages/_editor_guide_text.html.erb [hunk 1]
```diff
@@ -148,6 +148,17 @@
...
+      
+    <h3><strong>JSFiddle Embed</strong></h3>
+    <p>All you need is the full JSFiddle <code>link</code>, ending in the fiddle ID code, as follows:</p>
+    <code>{% jsfiddle https://jsfiddle.net/link2twenty/v2kx9jcd %}</code>
+
+    <dl>
+    <dt><code>Custom tabs</code></dt>
+    <dd>
+      You can add a custom tab order to you JSFiddle embed tag. Defaults to <i>js,html,css,result</i><br/>
+      <code>{% jsfiddle https://codepen.io/twhite96/pen/XKqrJX result,html,css %}</code>
```
Comment (Zhao-Andy) [inline app/views/pages/_editor_guide_text.html.erb]: Let's change it to a JSFiddle example, too.
```suggestion
      <code>{% jsfiddle https://jsfiddle.net/webdevem/Q8KVC result,html,css %}</code>
```

File: spec/liquid_tags/jsfiddle_tag_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    it "rejects invalid jsfiddle link" do
+      expect do
+        generate_new_liquid("invalid_jsfiddle_link")
+      end.to raise_error(StandardError)
+    end
+
+    it "accepts jsfiddle link with a custom-tab parameter" do
+      expect do
+        generate_new_liquid(jsfiddle_link_with_custom_tab)
```
Comment (benhalpern) [inline spec/liquid_tags/jsfiddle_tag_spec.rb]:
```suggestion
        generate_new_liquid(jsfiddle_link_with_custom_tabs)
```

Looks like a typo here

**Comment by Link2Twenty** [comment]
@benhalpern I think that should be everything.
I've given it another looking over and can't see anything else.

**Comment by benhalpern** [comment]
@Link2Twenty only problem left is a failing test

**Comment by Link2Twenty** [comment]
I've changed the test logic to an "if it doesn't give an error" type rather than an "if HTML matches template". This makes the tests pass but I'm not really happy doing this.

Is someone able to look at my approve.html and tell me what I've done wrong?

**Comment by maestromac** [comment]
Dealing with approvals spec can be frustrating on both local machines and on Travis. I've added [an additional step](https://github.com/thepracticaldev/dev.to/commit/b2bdb432060c75fdac2b36e13150c7b8c21a29d4) to Travis on master to print out the mismatch. would you like to revert some of your spec changes, merge in master, and see what would be outputted on Travis?

**Comment by Link2Twenty** [comment]
@maestromac looks like travis isn't firing now
https://travis-ci.com/thepracticaldev/dev.to/pull_requests

**Comment by maestromac** [comment]
let me try to re-trigger

**Comment by Link2Twenty** [comment]
@maestromac ah there's an error

"Could not parse .travis.yml"

https://travis-ci.com/thepracticaldev/dev.to/requests

**Comment by maestromac** [comment]
Thank you for catching that 😅

**Comment by Link2Twenty** [comment]
Of course `=""` was what I needed...

Well we've made it now 🙂

**Comment by Link2Twenty** [comment]
Thank you all for being patient with me 🙂

**Pull request merged**